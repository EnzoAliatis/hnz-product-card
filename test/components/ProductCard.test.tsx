import React from 'react';
import renderer from 'react-test-renderer';
import { ProductCard } from '../../src/components';
import { product } from '../data/products';

const { act } = renderer;

describe('ProductCart', () => {
  test('should show the component well mounted', () => {
    const wrapper = renderer.create(
      <ProductCard product={product} className="custom-class">
        {() => <h1>Product Cart</h1>}
      </ProductCard>
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('the counter should work', () => {
    const wrapper = renderer.create(
      <ProductCard product={product} className="custom-class">
        {({ count, increaseBy }) => (
          <>
            <h1>Product Card</h1>
            <span>{count}</span>
            <button onClick={() => increaseBy(3)}>+3</button>
          </>
        )}
      </ProductCard>
    );

    let tree = wrapper.toJSON();

    expect(tree).toMatchSnapshot();

    act(() => {
      (tree as any).children[2].props.onClick();
    });

    tree = wrapper.toJSON();

    expect((tree as any).children[1].children[0]).toBe('3');
  });
});
