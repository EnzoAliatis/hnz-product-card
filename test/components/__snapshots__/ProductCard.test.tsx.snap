// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCart should show the component well mounted 1`] = `
<div
  className="productCard custom-class"
>
  <h1>
    Product Cart
  </h1>
</div>
`;

exports[`ProductCart the counter should work 1`] = `
<div
  className="productCard custom-class"
>
  <h1>
    Product Card
  </h1>
  <span>
    0
  </span>
  <button
    onClick={[Function]}
  >
    +3
  </button>
</div>
`;
